lib/Doctrine/ORM/Tools/EntityGenerator.php

```diff
@@ -1876,7 +1876,7 @@ protected function prefixCodeWithSpaces($code, $num = 1)
         $lines = explode("\n", $code);
 
         foreach ($lines as $key => $value) {
-            if (! empty($value)) {
+            if (($value !== '' && $value !== '0')) {
```

Comment:
```suggestion
            if ($value !== '' && $value !== '0') {
```
Are you sure we want to test for `'0'` here?

Reply:
That's what empty() does under the hood. I can remove the '0' test, but this may have unintended consequences. (That's exactly why I despise empty() :D )

Reply:
Yeah me too, but now with your patch, people might think this `$value !== '0'` really is useful, and it might very well not be.

Reply:
What's best here, risk the side effect or revert?

Reply:
Reading the code and understanding what it does is best. Here, we have a method that indents a code block. Except it should not indent empty lines otherwise that will mean trailing spaces. So IMO we can drop that `&& $value !== '0'`, because in the super unlikely case we have a `0` sitting by itself on a single line in a code block, we want to indent it.